src/Service/OpenCatalogiService.php

```diff
@@ -0,0 +1,248 @@
+
+        return $organization;
+
+    }//end enrichOpencatalogiOrg()
+
+
+    /**
+     * This function loops through the array with publiccode/opencatalogi files.
+     *
+     * @param array        $opencatalogiArray The opencatalogi array from the github api.
+     * @param Source       $source            The github api source.
+     * @param ObjectEntity $repository        The repository object.
+     *
+     * @return ObjectEntity|null The repository with the updated organization object
+     * @throws Exception
+     */
+    public function handleOpencatalogiFile(array $opencatalogiArray, Source $source, ObjectEntity $repository, array $data, ?array $organizationArray=[]): ?ObjectEntity
+    {
+        $opencatalogiMapping = $this->resourceService->getMapping($this->configuration['opencatalogiMapping'], 'open-catalogi/open-catalogi-bundle');
+        $organizationSchema  = $this->resourceService->getSchema($this->configuration['organizationSchema'], 'open-catalogi/open-catalogi-bundle');
+        if ($opencatalogiMapping instanceof Mapping === false
+            || $organizationSchema instanceof Entity === false
+        ) {
+            return null;
+        }
+
+        $opencatalogi = $data['opencatalogi'];
+        $sha          = $data['sha'];
+
+        if ($repository->getValue('source') === 'gitlab') {
+            // Find the sync with the source and organization web_url.
+            $organizationSync       = $this->syncService->findSyncBySource($source, $organizationSchema, $organizationArray['web_url']);
+            $opencatalogi['gitlab'] = $organizationArray['web_url'];
+
+            if ($organizationArray['kind'] === 'group') {
+                $opencatalogi['type'] = 'Organization';
+            }
+
+            if ($organizationArray['kind'] === 'user') {
+                $opencatalogi['type'] = 'User';
+            }
+        }
+
+        if ($repository->getValue('source') === 'github') {
+            // Find the sync with the source and organization html_url.
+            $organizationSync       = $this->syncService->findSyncBySource($source, $organizationSchema, $organizationArray['html_url']);
+            $opencatalogi['github'] = $organizationArray['html_url'];
+            $opencatalogi['type']   = $organizationArray['type'];
+        }
+
+        // Check the sha of the sync with the url reference in the array.
+        if ($this->syncService->doesShaMatch($organizationSync, $sha) === true) {
+            return $repository;
+        }
+
+        // Set the mapping to the sync object.
+        $organizationSync->setMapping($opencatalogiMapping);
+
+        // Synchronize the organization with the opencatalogi file.
+        $organizationSync = $this->syncService->synchronize($organizationSync, $opencatalogi);
+
+        // Persist the organization sync and organization objects.
+        $this->entityManager->persist($organizationSync);
+        $this->entityManager->persist($organizationSync->getObject());
```

Comment:
You shouldn't need to do this
![image](https://github.com/OpenCatalogi/OpenCatalogiBundle/assets/3785488/d18fb5fa-31b8-4cce-b77a-69707e7a4c76)

However, it looks like we still try to set synchronization properties after persisting in the SyncService

Reply:
It shouldn't be necessary, but there are cases that somehow the object is not stored correctly if not persisted separately

Reply:
Weird, but oke. We should be fixing this when rewriting syncService one day